VirtoCommerce.Storefront/Domain/Cart/CartConverter.cs

```diff
@@ -429,13 +429,9 @@ public static ShoppingCart ToShoppingCart(this cartDto.ShoppingCart cartDto, Cur
                 Customer = user
             };
 
-            if (cartDto.Coupon != null)
+            if (cartDto.Coupons != null)
             {
-                result.Coupon = new Coupon
-                {
-                    Code = cartDto.Coupon,
-                    AppliedSuccessfully = !string.IsNullOrEmpty(cartDto.Coupon)
-                };
+                result.Coupons = cartDto.Coupons?.Select(c => new Coupon { Code = c, AppliedSuccessfully = !string.IsNullOrEmpty(c) }).ToList();
```

Comment:
You have checked `cartDto.Coupons` for null already, so you don't need to use `?.`